### DISCUSSION TITLE
2018 年总结 · /2018-summary

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#540] [Announcements]
https://draveness.me/2018-summary

**Comment by andyRon** [comment]
一开始，还以为“作者”是某个引用文章的的作者呢！🤦‍♀️🤦‍♀️

**Comment by draveness** [comment]
> @andyRon
> 一开始，还以为“作者”是某个引用文章的的作者呢！🤦‍♀️🤦‍♀️

不习惯在文章中用『我』，看了下全篇就几个

**Comment by FreeMrQ** [comment]
看的赏心悦目😄，顺便请教下2018 技术路线那个图是用什么软件做的，感觉很精致！🙏

**Comment by draveness** [comment]
> @Lance-Robot
> 看的赏心悦目😄，顺便请教下2018 技术路线那个图是用什么软件做的，感觉很精致！🙏

文章最后有

**Comment by SeanCST** [comment]
个人只是体系 -> 个人知识体系

**Comment by draveness** [comment]
> @SeanCST
> 个人只是体系 -> 个人知识体系

fixed

**Comment by authhwang** [comment]
作者现在能透漏现在在哪家公司工作吗？

**Comment by wahusky** [comment]
学习了，干货满满，感谢作者

**Comment by BeihaoZhang** [comment]
有几个老生常谈的问题，真诚地向作者请教一下，希望能得到作者的解惑。万分感谢😄😄😄
1. 关于学习这块，对于一块知识或者一门技术的学习和应用，如何拿捏好它的深度呢，有时候会一直盯着这块知识点，而忽略了以后看来其他真正重要的东西，这该如何避免？

2. 另外，对于一本书的好坏怎么评判呢，有时我读到一本我看不懂的书时，我不知道它是好书还是坏书？潜意识里总觉得自己看不懂的觉得高大上的书都是很牛逼的书。

3. 对于一门新技术的学习方向的把握，作者一般是怎么控制的呢？如何才能不跑偏，把握好哪些东西该深究哪些东西只要学会应用就好了。（看起来就是对第一个问题的补充😄）

4. 遇到学习障碍了，感到各种困惑，该怎么调节自己的状态呢，应该怎样去认识这个问题？

**Comment by draveness** [comment]
> @BeihaoZhang
> 有几个老生常谈的问题，真诚地向作者请教一下，希望能得到作者的解惑。万分感谢😄😄😄
> 1. 关于学习这块，对于一块知识或者一门技术的学习和应用，如何拿捏好它的深度呢，有时候会一直盯着这块知识点，而忽略了以后看来其他真正重要的东西，这该如何避免？
> 
> 2. 另外，对于一本书的好坏怎么评判呢，有时我读到一本我看不懂的书时，我不知道它是好书还是坏书？潜意识里总觉得自己看不懂的觉得高大上的书都是很牛逼的书。
> 
> 3. 对于一门新技术的学习方向的把握，作者一般是怎么控制的呢？如何才能不跑偏，把握好哪些东西该深究哪些东西只要学会应用就好了。（看起来就是对第一个问题的补充😄）
> 
> 4. 遇到学习障碍了，感到各种困惑，该怎么调节自己的状态呢，应该怎样去认识这个问题？

1. 对于经常使用的技术，不仅要深入理解这些内容，还要对 N + 1 层的知识进行掌握，平时用的各种工具（N + 1层）的设计思想和实现原理都是需要掌握的，最佳的办法是根据重要性对需要学习的东西进行排序，然后在一定时间内（三个月）做到对某个知识（编程语言、数据库、开源框架、网络等）设计思想和源码级别的掌握；
2. 如果觉得一本书讲的云里雾里根本看不懂，那其实没有必要读下去，大部分的书写的都不怎么样，尽量减少浪费的时间，这些时间可以用来看看国外大学科班使用的一些教材，不过如果真的有**知名的作者背书**，例如：Martin Fowler 这种级别的，其实有两种选择：
    1. 这种书大都介绍的不是具体的技术细节，如果时间不是特别充裕，可以每隔一段时间（半年、一年）随便翻一翻，总有一天会觉得这本书讲的东西总结了一些具体的知识形成了抽象的思想；
    2. 如果是比较成熟的工程师，可以强行啃下来，然后隔一段时间多啃一次，虽然很困难，但是绝对是有好处的，这些东西会潜移默化影响我们的思考方式；
3. 略；
4. 之前确实遇到过很多瓶颈期，觉得很久都没有进步，这时候应该多看看以前用来垫显示器的书，把不是特别清楚的原理想明白，我们需要清楚的是**技术的提升有的时候不是线性的**，就是一些关键点想明白看很多东西都会变清楚，作者在刚学习 OOP 的时候并不是说慢慢了解了 OOP 的几个原理和使用方法，而是积累了比较长的时间有一天突然就想明白了封装、继承和多态这些都是什么。

说了这么多，最重要的还是**多学多想**，作者所学还是有限，这里的回答也都是基于体验和经历，会有一些主观的因素在，不一定会对你有所帮助，不过非常感谢你提的这个问题，也让作者对自己的学习方式进行了反思。

**Comment by BeihaoZhang** [comment]
非常感谢作者的细心解答！！！

**Comment by XhinLiang** [comment]
恭喜转职成功。

**Comment by suibber** [comment]
关注，希望作者多多更新博客

**Comment by g10guang** [comment]
请问在哪里工作？好像工作内容很酷的样子

**Comment by OC0311** [comment]
看了您的文章有两个问题想要请教一下
1、您是一段时间只会专研一个技术吗？
2、我平常很少去深入研究一些新技术，但是会认真去看一些经典的基础书籍如现代操作系统,TCP/IP等着一类书，但是基本上看完就忘了，需要反复的翻阅，但是又感觉这样在工作中给我带来的成长不是很明显，请问有必要继续吗？

希望能得到您的答复

**Comment by draveness** [comment]
这些问题都挺主观的，因人而异吧

> 1、您是一段时间只会专研一个技术吗？

差不多，不会特别发散，但是也不会说只看一个东西

> 2、我平常很少去深入研究一些新技术，但是会认真去看一些经典的基础书籍如现代操作系统,TCP/IP等着一类书，但是基本上看完就忘了，需要反复的翻阅，但是又感觉这样在工作中给我带来的成长不是很明显，请问有必要继续吗？

+ 如果没看过或者不是 CS 相关专业的，可以学一遍；
+ 如果看过了或者学过了，再翻阅觉得没啥成长就没有必要读了，真觉得有需要的时候再看也可以吧
